text/0569-tracked-list.md

```diff
@@ -0,0 +1,306 @@
+- Start Date: 2019-12-23
+- Relevant Team(s): Ember.js
+- RFC PR: https://github.com/emberjs/rfcs/pull/569
+- Tracking: (leave this empty)
+
+# TrackedList
+
+## Summary
+
+Introduces a new array-like class whose usage is autotracked by default, and
+is designed with performance in mind.
+
+```js
+import { TrackedList } from '@glimmer/tracking';
+
+class Person {
+  friends = TrackedList.from(['Yehuda', 'Melanie', 'Ricardo']);
+
+  addFriend(newFriend) {
+    this.friends.push(newFriend);
+  }
+}
+```
+
+## Motivation
```

Comment:
I'd add to this that there's another (and IMO even more important than paradigm-preference) reason that this is valuable: performance. Immutable functional updates have many advantages, but without something like [persistent data structures](https://en.wikipedia.org/wiki/Persistent_data_structure), they can carry  a high performance penalty.

`TrackedList` allows users to make that tradeoff as they like, whatever their programming paradigm preference—and there's no reason that you can't provide a functional abstraction around it if that's your jam.